Add "Back up data" step to UA

cjcenizal opened the pull request:
This PR adds the on-prem state for this step to Upgrade Assistant. Because I'm linking to Snapshot and Restore, this PR also adds a locator to the `snapshot_restore` plugin, which I'll forward-port to master.

![image](https://user-images.githubusercontent.com/1238659/130286281-5805c1b3-a991-44ad-bbae-fa8585e09238.png)

I'll add tests once Alison's CIT refactor is merged.

**To-do**
- [x] CITs
- [x] On-Cloud state https://github.com/elastic/kibana/pull/109956
- [x] Forward-port SR locator to master (https://github.com/elastic/kibana/pull/109886)

elasticmachine commented:
Pinging @elastic/kibana-stack-management (Team:Stack Management)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/)
* Commit: 8bc82cdde73ab66991e97923bffff71e9a07fe28
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/execution/node/422/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/)
* Commit: f3636ddc874e31ff3a81c64c2fcc51e74df2e00a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/execution/node/428/log)

### History
* :broken_heart: [Build #147499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/) failed 8bc82cdde73ab66991e97923bffff71e9a07fe28

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/)
* Commit: f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/execution/node/426/log)

### History
* :broken_heart: [Build #147515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/) failed f3636ddc874e31ff3a81c64c2fcc51e74df2e00a
* :broken_heart: [Build #147499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/) failed 8bc82cdde73ab66991e97923bffff71e9a07fe28

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"}]}
PIPELINE-->

cjcenizal commented:
@alisonelizabeth I ran the tests and the TS check locally and everything passed. I think this is ready for review.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/)
* Commit: 7ac053c96964e5a982275aa72acd5b86cd1cd21d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109543/7ac053c96964e5a982275aa72acd5b86cd1cd21d)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/7ac053c96964e5a982275aa72acd5b86cd1cd21d) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 180 | 181 | +1 |
| `upgradeAssistant` | 126 | 127 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/7ac053c96964e5a982275aa72acd5b86cd1cd21d) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 453.5KB | 450.0KB | -3.6KB |
| `upgradeAssistant` | 196.6KB | 198.4KB | +1.8KB |
| total |  |  | -1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/7ac053c96964e5a982275aa72acd5b86cd1cd21d) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 41.4KB | 47.3KB | +5.9KB |
| `upgradeAssistant` | 18.5KB | 18.5KB | +18.0B |
| total |  |  | +5.9KB |

### History
* :broken_heart: [Build #147937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/) failed f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* :broken_heart: [Build #147515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/) failed f3636ddc874e31ff3a81c64c2fcc51e74df2e00a
* :broken_heart: [Build #147499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/) failed 8bc82cdde73ab66991e97923bffff71e9a07fe28

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/","number":"147918","commit":"7ac053c96964e5a982275aa72acd5b86cd1cd21d"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/)
* Commit: 4fc94075fd2efd9b717c6f3797a19f644df58869
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109543/4fc94075fd2efd9b717c6f3797a19f644df58869)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/4fc94075fd2efd9b717c6f3797a19f644df58869) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 180 | 181 | +1 |
| `upgradeAssistant` | 126 | 127 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/4fc94075fd2efd9b717c6f3797a19f644df58869) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 453.5KB | 450.0KB | -3.6KB |
| `upgradeAssistant` | 196.6KB | 198.4KB | +1.8KB |
| total |  |  | -1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/4fc94075fd2efd9b717c6f3797a19f644df58869) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 41.4KB | 47.3KB | +5.9KB |
| `upgradeAssistant` | 18.5KB | 18.5KB | +18.0B |
| total |  |  | +5.9KB |

### History
* :broken_heart: [Build #147937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/) failed f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* :green_heart: [Build #147918](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/) succeeded 7ac053c96964e5a982275aa72acd5b86cd1cd21d
* :broken_heart: [Build #147515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/) failed f3636ddc874e31ff3a81c64c2fcc51e74df2e00a
* :broken_heart: [Build #147499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/) failed 8bc82cdde73ab66991e97923bffff71e9a07fe28

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/","number":"147918","commit":"7ac053c96964e5a982275aa72acd5b86cd1cd21d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/","number":"147961","commit":"4fc94075fd2efd9b717c6f3797a19f644df58869"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/backup_step/backup_step.tsx:
```diff
@@ -0,0 +1,78 @@
+
+  return (
+    <EuiButton href={snapshotRestoreUrl} data-test-subj="snapshotRestoreLink">
+      <FormattedMessage
```
[nit] can you move this to `i18nTexts` for consistency?

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/backup_step/backup_step.tsx:
```diff
@@ -0,0 +1,78 @@
+  }),
+
+  backupStepDescription: i18n.translate('xpack.upgradeAssistant.overview.backupStepDescription', {
+    defaultMessage: 'Back up your data before addressing any deprecation issues.',
```
I'll defer to @debadair, but I think in some places we've changed the copy to "deprecation warnings" rather than "deprecation issues".

cjcenizal replied:
Makes sense. I did a quick search of the codebase and found numerous references to "deprecation issues" outside the scope of this PR. How about I address them all in their own PR?

alisonelizabeth replied:
👍 sounds good, thanks!

alisonelizabeth commented on x-pack/plugins/upgrade_assistant/public/application/mount_management_section.tsx:
```diff
@@ -5,19 +5,36 @@
+import { render, unmountComponentAtNode } from 'react-dom';
 import { CoreSetup } from 'src/core/public';
-import { ManagementAppMountParams } from '../../../../../src/plugins/management/public';
-import { renderApp } from './render_app';
```
Can this file be removed now?

alisonelizabeth reviewed (commented):
Thanks for working on this @cjcenizal! Tested locally and works as expected. Left a couple minor comments in the code.

I also have a question around the UX. I'm not sure about having SR open in the same tab. If we had the ability in SR to navigate the user back to UA then I would agree with the current approach. However, since we don't, it seems like it would be better to open SR in a new tab, so that the user can easily switch back to UA after creating a snapshot. WDYT?

alisonelizabeth commented:
@cjcenizal also, forgot to ask - will the on-cloud state be handled in a separate PR?

cjcenizal commented:
@alisonelizabeth Thanks for the review! I addressed your comments except for the copy one, which I think would be better to handle separately. To your question about the on-Cloud state, yes, I'll implement that in a subsequent PR. Trying to keep my PRs as small as possible.

> However, since we don't, it seems like it would be better to open SR in a new tab, so that the user can easily switch back to UA after creating a snapshot.

I think you raise a good question. I haven't seen guidelines from the designers on whether to default to opening in-app links in the same window or in a new one. Personally, I believe the back button and side nav are both reasonable options for navigating back to UA, and that defaulting to opening links in the same tab preserves the option of right-clicking links to open them in new ones, which I prefer over restricting folks to new tabs only. How about I follow up with the design team for guidance, and we retain the current implementation until we have a clear direction?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/)
* Commit: ff9780313c68d16fd8e950585256e372d6912d13
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109543/ff9780313c68d16fd8e950585256e372d6912d13)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/ff9780313c68d16fd8e950585256e372d6912d13) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 180 | 181 | +1 |
| `upgradeAssistant` | 126 | 127 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/ff9780313c68d16fd8e950585256e372d6912d13) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 453.5KB | 450.0KB | -3.6KB |
| `upgradeAssistant` | 196.6KB | 198.3KB | +1.8KB |
| total |  |  | -1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/ff9780313c68d16fd8e950585256e372d6912d13) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 41.4KB | 47.3KB | +5.9KB |
| `upgradeAssistant` | 18.5KB | 18.5KB | +18.0B |
| total |  |  | +5.9KB |

### History
* :green_heart: [Build #147961](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/) succeeded 4fc94075fd2efd9b717c6f3797a19f644df58869
* :broken_heart: [Build #147937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/) failed f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* :green_heart: [Build #147918](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/) succeeded 7ac053c96964e5a982275aa72acd5b86cd1cd21d
* :broken_heart: [Build #147515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/) failed f3636ddc874e31ff3a81c64c2fcc51e74df2e00a
* :broken_heart: [Build #147499](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/) failed 8bc82cdde73ab66991e97923bffff71e9a07fe28

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147499/","number":"147499","commit":"8bc82cdde73ab66991e97923bffff71e9a07fe28"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/","number":"147918","commit":"7ac053c96964e5a982275aa72acd5b86cd1cd21d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/","number":"147961","commit":"4fc94075fd2efd9b717c6f3797a19f644df58869"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/","number":"148004","commit":"ff9780313c68d16fd8e950585256e372d6912d13"}]}
PIPELINE-->

sabarasaba commented on x-pack/plugins/upgrade_assistant/__jest__/client_integration/overview/backup_step/backup_step.test.tsx:
```diff
@@ -0,0 +1,30 @@
+  });
+
+  describe('On-prem', () => {
+    test('Shows link to Snapshot and Restore', () => {
```
Was wondering if we should also assert on the url of the link in order to be more explicit about where we are expecting users to go 🤔

cjcenizal replied:
Thanks for suggesting this! I added a test, but tried to structure the mocks to avoid testing the locator itself. Love to hear your thoughts on this.

sabarasaba commented:
> How about I follow up with the design team for guidance, and we retain the current implementation until we have a clear direction?

@cjcenizal Once we hear back from them we might also want to update the links to Observability and Discover apps since at the moment they open in new tabs

sabarasaba reviewed (approved):
Thanks for taking care of this @cjcenizal! Tested locally and everything works as it should. I left one minor nit, but feel free to ignore.

alisonelizabeth commented:
> I think you raise a good question. I haven't seen guidelines from the designers on whether to default to opening in-app links in the same window or in a new one. Personally, I believe the back button and side nav are both reasonable options for navigating back to UA, and that defaulting to opening links in the same tab preserves the option of right-clicking links to open them in new ones, which I prefer over restricting folks to new tabs only. How about I follow up with the design team for guidance, and we retain the current implementation until we have a clear direction?

You bring up some good points. I would argue the back button isn't ideal, especially since you may need to go through various flows in SR to create a snapshot (set up repo, create policy, run policy, verify snapshot). But I agree the side nav is an option. Also, I agree that the user is still able to open in a new tab via right-click (although IMO, when I see a button without the external icon, I tend to think it is within the app's flow so I might not think about opening it in a new tab).

I'm OK with following up on this if you'd like to get this merged. As @sabarasaba pointed out, we should change the Observability and Discover links if this is the direction we want to take. Also, FWIW, in ILM, we also open in an external tab, so it'd be great to be consistent across our apps. 

<img width="471" alt="Screen Shot 2021-08-24 at 9 03 26 AM" src="https://user-images.githubusercontent.com/5226211/130622069-14c9021a-982b-4e03-8c37-7dff50512d04.png">

Curious on @dborodyansky's thoughts as well, as the prototype appears it would open in a separate tab (with external icon).

<img width="499" alt="Screen Shot 2021-08-24 at 9 06 10 AM" src="https://user-images.githubusercontent.com/5226211/130622185-9425143c-ba2a-4767-b5fd-c05932d38c68.png">

alisonelizabeth reviewed (approved):
Latest LGTM. Responded to your comment around the "Create snapshot" button, but happy to follow up in a separate PR.

I also opened https://github.com/elastic/kibana/issues/109834 to update the SR links in ILM once your changes are forward-ported to master.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148259/)
* Commit: 993ca8faec8097c0c929f37be1a5cd65a4ae7fd5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109543/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 180 | 181 | +1 |
| `upgradeAssistant` | 126 | 127 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 453.5KB | 450.0KB | -3.6KB |
| `upgradeAssistant` | 196.6KB | 198.3KB | +1.8KB |
| total |  |  | -1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 41.4KB | 47.3KB | +5.9KB |
| `upgradeAssistant` | 18.5KB | 18.5KB | +18.0B |
| total |  |  | +5.9KB |

### History
* :green_heart: [Build #148004](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/) succeeded ff9780313c68d16fd8e950585256e372d6912d13
* :green_heart: [Build #147961](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/) succeeded 4fc94075fd2efd9b717c6f3797a19f644df58869
* :broken_heart: [Build #147937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/) failed f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* :green_heart: [Build #147918](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/) succeeded 7ac053c96964e5a982275aa72acd5b86cd1cd21d
* :broken_heart: [Build #147515](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/) failed f3636ddc874e31ff3a81c64c2fcc51e74df2e00a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147515/","number":"147515","commit":"f3636ddc874e31ff3a81c64c2fcc51e74df2e00a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/","number":"147918","commit":"7ac053c96964e5a982275aa72acd5b86cd1cd21d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/","number":"147961","commit":"4fc94075fd2efd9b717c6f3797a19f644df58869"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/","number":"148004","commit":"ff9780313c68d16fd8e950585256e372d6912d13"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148259/","number":"148259","commit":"993ca8faec8097c0c929f37be1a5cd65a4ae7fd5"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148363/)
* Commit: 993ca8faec8097c0c929f37be1a5cd65a4ae7fd5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109543/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 180 | 181 | +1 |
| `upgradeAssistant` | 126 | 127 | +1 |
| total |  |  | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 453.5KB | 450.0KB | -3.6KB |
| `upgradeAssistant` | 196.6KB | 198.3KB | +1.8KB |
| total |  |  | -1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfa1e83e94f85205bc2743dc928e3e7530f37950) | [after](https://github.com/elastic/kibana/commit/993ca8faec8097c0c929f37be1a5cd65a4ae7fd5) | diff |
| --- | --- | --- | --- |
| `snapshotRestore` | 41.4KB | 47.3KB | +5.9KB |
| `upgradeAssistant` | 18.5KB | 18.5KB | +18.0B |
| total |  |  | +5.9KB |

### History
* :green_heart: [Build #148259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148259/) succeeded 993ca8faec8097c0c929f37be1a5cd65a4ae7fd5
* :green_heart: [Build #148004](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/) succeeded ff9780313c68d16fd8e950585256e372d6912d13
* :green_heart: [Build #147961](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/) succeeded 4fc94075fd2efd9b717c6f3797a19f644df58869
* :broken_heart: [Build #147937](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/) failed f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c
* :green_heart: [Build #147918](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/) succeeded 7ac053c96964e5a982275aa72acd5b86cd1cd21d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147918/","number":"147918","commit":"7ac053c96964e5a982275aa72acd5b86cd1cd21d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147937/","number":"147937","commit":"f3ea2ff6d18db17aaafcf4773943e2f7b85ebf5c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147961/","number":"147961","commit":"4fc94075fd2efd9b717c6f3797a19f644df58869"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148004/","number":"148004","commit":"ff9780313c68d16fd8e950585256e372d6912d13"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148259/","number":"148259","commit":"993ca8faec8097c0c929f37be1a5cd65a4ae7fd5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148363/","number":"148363","commit":"993ca8faec8097c0c929f37be1a5cd65a4ae7fd5"}]}
PIPELINE-->
